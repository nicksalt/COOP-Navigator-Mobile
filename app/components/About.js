import React from "react";
import { Text, View, ScrollView, Linking, StyleSheet } from "react-native";
import Field from "./Field";

export default class About extends React.Component {
  render() {
    return (
      <ScrollView>
        <Text style={styles.subtitle}>General Information</Text>
        <Text style={styles.aboutText}>
          Co-operative education at the University of Ottawa allows you to apply
          concepts learned in class during paid work terms. After about four
          years of study, you will have not only a diploma that indicates you
          participated in a CO-OP program but also approximately 16 months of
          experience in your field of study and a network of valuable contacts.
          All of these factors will contribute to helping you find a job more
          easily after graduation.
        </Text>
        <Text style={styles.subtitle}>Contact Information</Text>
        <Field
          field="Email"
          fieldValue={
            <Text
              style={styles.link}
              onPress={() => Linking.openURL("mailto:coop@uottawa.ca")}
            >
              coop@uottawa.ca
            </Text>
          }
        />
        <Field
          field="Telephone"
          fieldValue={
            <Text
              style={styles.link}
              onPress={() => Linking.openURL("tel:613-562‑5741")}
            >
              613-562‑5741
            </Text>
          }
        />
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  aboutText: {
    padding: 20,
    fontWeight: "bold",
    color: "black",
    textAlign: "justify"
  },
  subtitle: {
    padding: 10,
    fontSize: 16,
    fontWeight: "bold",
    textAlign: "center",
    color: "white",
    backgroundColor: "#a9343a"
  },
  link: {
    color: "blue",
    textDecorationLine: "underline",
    fontSize: 18
  }
});
