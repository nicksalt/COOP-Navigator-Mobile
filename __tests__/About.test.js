import React from "react";
import About from "../app/components/About";
import renderer from "react-test-renderer";

jest.mock("ScrollView", () => "ScrollView");

test("About snapshot renders correctly", () => {
  const tree = renderer.create(<About />).toJSON();
  expect(tree).toMatchSnapshot();
});
