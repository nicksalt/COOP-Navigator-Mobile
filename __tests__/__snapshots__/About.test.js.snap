// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About snapshot renders correctly 1`] = `
<ScrollView>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    General Information
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontWeight": "bold",
        "padding": 20,
        "textAlign": "justify",
      }
    }
  >
    Co-operative education at the University of Ottawa allows you to apply concepts learned in class during paid work terms. After about four years of study, you will have not only a diploma that indicates you participated in a CO-OP program but also approximately 16 months of experience in your field of study and a network of valuable contacts. All of these factors will contribute to helping you find a job more easily after graduation.
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Contact Information
  </Text>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      Email
    </Text>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 18,
            "textDecorationLine": "underline",
          }
        }
      >
        coop@uottawa.ca
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 10,
        }
      }
    >
      Telephone
    </Text>
    <View
      style={Object {}}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "color": "blue",
            "fontSize": 18,
            "textDecorationLine": "underline",
          }
        }
      >
        613-562â€‘5741
      </Text>
    </View>
  </View>
</ScrollView>
`;
